### PR TITLE
API-002 Adds unit tests to OrderFactory and Order

### PR DESCRIPTION
**Pull request by jakubgiminski**
Task: https://github.com/jakubgiminski/order-item/projects/1#card-5261477

### CODE REVIEW

File: tests/Unit/Order/OrderFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace OrderItemTest\Unit\Order;
+
+use OrderItem\Item\Item;
+use OrderItem\Item\ItemsCollection;
+use OrderItem\Order\Order;
+use OrderItem\Order\OrderFactory;
+use PHPUnit\Framework\TestCase;
+use Ramsey\Uuid\Uuid;
+
+class OrderFactoryTest extends TestCase
```
Comment (malexc) [inline tests/Unit/Order/OrderFactoryTest.php]: Additional test required for no items.

File: src/Order/Order.php [hunk 2]
```diff
@@ -24,6 +24,11 @@ public function getId(): string
         return $this->id;
     }
 
+    public function setItems(ItemsCollection $items): void
```
Comment (dbojdo) [inline src/Order/Order.php]: ItemsCollection should be passed in the constructor. Order is an aggregate and should manage its items via relevant methods (add / remove item).

File: tests/Unit/Order/OrderTest.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+use PHPUnit\Framework\TestCase;
+use Ramsey\Uuid\Uuid;
+
+class OrderTest extends TestCase
+{
+    public function testCreateOrderWithoutItems()
+    {
+        $order = new Order(Uuid::uuid1()->toString());
+
+        self::assertInstanceOf(Order::class, $order);
```
Comment (dbojdo) [inline tests/Unit/Order/OrderTest.php]: This assertion is meaningless

File: tests/Unit/Order/OrderTest.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public function testAddItemsToOrder()
+    {
+        $order = new Order(Uuid::uuid1()->toString());
+        $numItems = 3;
+        $items = $this->createItems($numItems);
```
Comment (dbojdo) [inline tests/Unit/Order/OrderTest.php]: This is not relevant for this test. It should check only if the item is added correctly, not need to setItems before.

**Pull request closed without merging**